### PR TITLE
Upgrade to latest core/contrib, remove usage of deprecated Int[Sum|Gauge]

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: tests/testutils/pdata_to_resource_metrics.go [hunk 1]
```diff
@@ -85,7 +79,7 @@ func addDoubleSum(ilms *InstrumentationLibraryMetrics, metric pdata.Metric) {
 	}
 	for l := 0; l < doubleSum.DataPoints().Len(); l++ {
 		dp := doubleSum.DataPoints().At(l)
-		val := dp.Value()
+		val := dp.DoubleVal()
```
Comment (rmfitzpatrick) [inline tests/testutils/pdata_to_resource_metrics.go:82]: I think this is `dp.Type()` dependent and `IntVal()` is needed for `MetricValueTypeInt`.

**Comment by rmfitzpatrick** [comment]
@bogdandrutu any chance you can run `make integration-test` and update w/ the results?  I'd be happy to push changes into your branch if preferred.

File: tests/testutils/pdata_to_resource_metrics.go (hunk 1 above)
Comment (bogdandrutu) [reply tests/testutils/pdata_to_resource_metrics.go]: Done.

File: tests/receivers/smartagent/collectd-tomcat/tomcat_test.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ var apache = []testutils.Container{
 	).WithExposedPorts("5000:5000").WithName("tomcat").WillWaitForPorts("5000"),
 }
 
-func TestCollectdApacheReceiverProvidesDefaultMetrics(t *testing.T) {
+func TestCollectdTomcatReceiverProvidesDefaultMetrics(t *testing.T) {
```
Comment (rmfitzpatrick) [inline tests/receivers/smartagent/collectd-tomcat/tomcat_test.go:30]: 👍

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**